server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
Pulp isn't really ready to work with a clustered MongoDB. It used to be with seeds, but then MongoDB approach to replication evolved and moved away from that. My understanding of these things is weak, so maybe check with @rbarlow who may have a better understanding of this stuff.

Focusing on just the goal of this PR, I'll suggest that instead of using `safe=True` you should pass the option `write_concern={fsync: True}` or `write_concern={w:1, fsync: True}`. The fsync option will perform a synchronous flush of all mongoDB writes which should guarantee that the write is committed to disk before returning. See the [fsync docs](http://docs.mongodb.org/v2.4/reference/command/fsync/) for more details. Note, in the 0.7.10 version it was not called `write_concern`, it was called `write_options`. You'll need this code to work correctly with all the versions that could be found on our platforms. See [this table](https://apps.fedoraproject.org/packages/python-mongoengine) for the list of versions on platforms.